File: src/services/mockData.ts
import { User, Barber, Service, Schedule, Plan, Membership, Notification, Review } from '../types';

// Helper to load/save from localStorage
const LOCAL_STORAGE_KEY = 'barber_osbao_mock_db';

export interface MockDatabase {
  currentUser: User;
  barbers: Barber[];
  services: Service[];
  plans: Plan[];
  memberships: Membership[];
  schedules: Schedule[];
  notifications: Notification[];
  reviews: Review[];
}

const INITIAL_USER: User = {
  id: 'usr-1',
  name: 'Fábio Zvir',
  email: 'fabio@barberosbao.com.br',
  phone: '(11) 98765-4321',
  avatarUrl: 'https://images.unsplash.com/photo-1535713875002-d1d0cf377fde?auto=format&fit=crop&q=80&w=150&h=150',
  preferences: {
    theme: 'dark',
    language: 'pt-BR',
    notifications: {
      email: true,
      push: true,
      whatsapp: true
    },
    privacy: {
      shareData: false,
      profilePublic: true
    }
  }
};

const INITIAL_SERVICES: Service[] = [
  {
    id: 'srv-1',
    name: 'Corte de Cabelo',
    category: 'cabelo',
    price: 45.00,
    durationMinutes: 30,
    iconName: 'scissors',
    description: 'Corte clássico ou moderno, adaptado ao seu estilo. Inclui lavagem simples.'
  },
  {
    id: 'srv-2',
    name: 'Barba',
    category: 'barba',
    price: 35.00,
    durationMinutes: 30,
    iconName: 'beard',
    description: 'Aparado, alinhamento com toalha quente e finalização com balm premium.'
  },
  {
    id: 'srv-3',
    name: 'Sobrancelha',
    category: 'sobrancelha',
    price: 20.00,
    durationMinutes: 15,
    iconName: 'eyebrow',
    description: 'Design de sobrancelha na pinça ou na navalha para realçar o olhar.'
  },
  {
    id: 'srv-4',
    name: 'Combo',
    category: 'combo',
    price: 90.00,
    durationMinutes: 75,
    iconName: 'combo',
    description: 'O pacote completo: Corte de Cabelo, Barba, Sobrancelha e uma massagem capilar.'
  }
];

const INITIAL_BARBERS: Barber[] = [
  {
    id: 'brb-1',
    name: 'Marcos Silva (Mestre)',
    avatarUrl: 'https://images.unsplash.com/photo-1500648767791-00dcc994a43e?auto=format&fit=crop&q=80&w=150&h=150',
    rating: 4.9,
    specialties: ['Corte Clássico', 'Fade/Degradê', 'Barba Terapia'],
    bio: 'Mais de 10 anos de experiência com tesoura e navalha. Especialista em visagismo.',
    availableDays: [], // Dynamically generated
    availableHours: ['09:00', '10:00', '11:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00', '19:00']
  },
  {
    id: 'brb-2',
    name: 'Thiago Neves',
    avatarUrl: 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?auto=format&fit=crop&q=80&w=150&h=150',
    rating: 4.8,
    specialties: ['Corte Moderno', 'Platinados', 'Desenho/Risco'],
    bio: 'Especialista em cortes urbanos e tendências modernas de penteado.',
    availableDays: [],
    availableHours: ['09:30', '10:30', '11:30', '13:30', '14:30', '15:30', '16:30', '17:30', '18:30']
  },
  {
    id: 'brb-3',
    name: 'Lucas Souza',
    avatarUrl: 'https://images.unsplash.com/photo-1628157582853-a796fa650a6a?auto=format&fit=crop&q=80&w=150&h=150',
    rating: 4.7,
    specialties: ['Barboterapia', 'Sobrancelha', 'Penteados'],
    bio: 'Foco no bem-estar e no relaxamento durante os cuidados com a barba.',
    availableDays: [],
    availableHours: ['09:00', '10:00', '11:00', '14:00', '15:00', '16:00', '17:00', '18:00']
  }
];

const INITIAL_PLANS: Plan[] = [
  {
    id: 'pln-1',
    name: 'Plano Cavalheiro',
    price: 89.90,
    period: 'mensal',
    benefits: [
      '2 Cortes de Cabelo por mês',
      '1 Design de Sobrancelha grátis',
      '10% de desconto em produtos',
      'Atendimento prioritário sem fila'
    ]
  },
  {
    id: 'pln-2',
    name: 'Plano Barão (Mais Popular)',
    price: 139.90,
    period: 'mensal',
    benefits: [
      'Cortes de Cabelo ILIMITADOS',
      '2 Serviços de Barba por mês',
      'Sobrancelha livre inclusa',
      '15% de desconto em produtos',
      '1 Cerveja ou café cortesia por visita'
    ],
    recommended: true
  },
  {
    id: 'pln-3',
    name: 'Plano Imperial',
    price: 199.90,
    period: 'mensal',
    benefits: [
      'Cortes e Barbas ILIMITADOS',
      'Todos os serviços extras inclusos (Sobrancelha, Selagem)',
      '20% de desconto em produtos',
      'Bebidas liberadas no bar (Cerveja/Café/Água)',
      'Acesso ao lounge VIP da barbearia'
    ]
  }
];

const INITIAL_NOTIFICATIONS: Notification[] = [
  {
    id: 'ntf-1',
    userId: 'usr-1',
    title: 'Bem-vindo à Barbearia Osbão!',
    message: 'Obrigado por criar sua conta. Explore nossos serviços e agende seu horário.',
    type: 'success',
    read: false,
    createdAt: new Date().toISOString()
  },
  {
    id: 'ntf-2',
    userId: 'usr-1',
    title: 'Clube do Assinante',
    message: 'Assine um de nossos planos e ganhe cortes ilimitados com descontos imperdíveis.',
    type: 'info',
    read: false,
    createdAt: new Date(Date.now() - 3600000 * 2).toISOString()
  }
];

const INITIAL_REVIEWS: Review[] = [
  {
    id: 'rev-1',
    scheduleId: 'sch-past-1',
    userId: 'usr-1',
    userName: 'Fábio Zvir',
    barberId: 'brb-1',
    rating: 5,
    comment: 'Atendimento excepcional do Marcos, corte perfeito e café maravilhoso.',
    createdAt: new Date(Date.now() - 3600000 * 24 * 18).toISOString()
  }
];

const INITIAL_SCHEDULES: Schedule[] = [
  {
    id: 'sch-past-1',
    userId: 'usr-1',
    barberId: 'brb-1',
    barberName: 'Marcos Silva (Mestre)',
    barberAvatar: 'https://images.unsplash.com/photo-1500648767791-00dcc994a43e?auto=format&fit=crop&q=80&w=150&h=150',
    services: [INITIAL_SERVICES[0]], // Corte
    date: new Date(Date.now() - 3600000 * 24 * 18).toISOString().split('T')[0], // 18 days ago
    time: '14:00',
    totalValue: 45.00,
    status: 'completed'
  }
];

// Helper to fill next 14 days availability for barbers
const populateBarberAvailability = (barbers: Barber[]): Barber[] => {
  const next14Days: string[] = [];
  for (let i = 0; i < 14; i++) {
    const d = new Date();
    d.setDate(d.getDate() + i);
    next14Days.push(d.toISOString().split('T')[0]);
  }
  return barbers.map(barber => ({
    ...barber,
    availableDays: next14Days
  }));
};

export const getInitialDB = (): MockDatabase => {
  const stored = localStorage.getItem(LOCAL_STORAGE_KEY);
  if (stored) {
    try {
      return JSON.parse(stored);
    } catch (e) {
      console.error('Failed to parse mock database, resetting', e);
    }
  }

  const db: MockDatabase = {
    currentUser: INITIAL_USER,
    barbers: populateBarberAvailability(INITIAL_BARBERS),
    services: INITIAL_SERVICES,
    plans: INITIAL_PLANS,
    memberships: [], // Initially no active plan as per mockup
    schedules: INITIAL_SCHEDULES,
    notifications: INITIAL_NOTIFICATIONS,
    reviews: INITIAL_REVIEWS
  };

  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(db));
  return db;
};

export const saveDB = (db: MockDatabase) => {
  localStorage.setItem(LOCAL_STORAGE_KEY, JSON.stringify(db));
};

File: lib/packages/core/utils/mock_data.dart
import 'package:barber_osbao/packages/core/models/user.dart';
import 'package:barber_osbao/packages/core/models/customer.dart';
import 'package:barber_osbao/packages/core/models/barber.dart';
import 'package:barber_osbao/packages/core/models/service_model.dart';
import 'package:barber_osbao/packages/core/models/appointment.dart';
import 'package:barber_osbao/packages/core/models/plan.dart';
import 'package:barber_osbao/packages/core/models/membership.dart';
import 'package:barber_osbao/packages/core/models/review.dart';
import 'package:barber_osbao/packages/core/models/notification_model.dart';

class MockData {
  MockData._();

  static final User loggedUser = User(
    id: 'usr_1',
    name: 'Fabio Zvir',
    email: 'fabio@barberosbao.com.br',
    phone: '(11) 99999-9999',
    avatarUrl: 'https://images.unsplash.com/photo-1534528741775-53994a69daeb?q=80&width=150',
    theme: 'dark',
    language: 'pt-BR',
    emailNotifications: true,
    pushNotifications: true,
    whatsappNotifications: false,
  );

  static final List<Barber> barbers = [
    const Barber(
      id: 'barb_1',
      name: 'Marcos Silva',
      avatarUrl: 'https://images.unsplash.com/photo-1517841905240-472988babdf9?q=80&width=150',
      rating: 4.9,
      specialties: ['Corte Clássico', 'Barboterapia', 'Degradê'],
      bio: 'Especialista em cortes clássicos e barboterapia com mais de 8 anos de experiência.',
      availableDays: ['2026-07-08', '2026-07-09', '2026-07-10', '2026-07-11', '2026-07-13', '2026-07-14'],
      availableHours: ['09:00', '10:00', '11:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00'],
    ),
    const Barber(
      id: 'barb_2',
      name: 'Arthur Santos',
      avatarUrl: 'https://images.unsplash.com/photo-1539571696357-5a69c17a67c6?q=80&width=150',
      rating: 4.8,
      specialties: ['Corte Moderno', 'Platinado', 'Design de Sobrancelha'],
      bio: 'Referência em cortes modernos, fade e colorimetria avançada.',
      availableDays: ['2026-07-08', '2026-07-09', '2026-07-10', '2026-07-11', '2026-07-13', '2026-07-14'],
      availableHours: ['09:00', '10:00', '11:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00'],
    ),
    const Barber(
      id: 'barb_3',
      name: 'Gabriel Neves',
      avatarUrl: 'https://images.unsplash.com/photo-1507003211169-0a1dd7228f2d?q=80&width=150',
      rating: 4.7,
      specialties: ['Corte na Tesoura', 'Barba Tradicional'],
      bio: 'Especialista em visagismo e finalizações clássicas na tesoura.',
      availableDays: ['2026-07-08', '2026-07-09', '2026-07-10', '2026-07-11', '2026-07-13', '2026-07-14'],
      availableHours: ['09:00', '10:00', '11:00', '13:00', '14:00', '15:00', '16:00', '17:00', '18:00'],
    ),
  ];

  static final List<ServiceModel> services = [
    const ServiceModel(
      id: 'srv_1',
      name: 'Corte de Cabelo',
      category: 'cabelo',
      price: 45.0,
      durationMinutes: 30,
      iconName: 'scissors',
      description: 'Corte completo com lavagem inclusa e finalização com pomada premium.',
    ),
    const ServiceModel(
      id: 'srv_2',
      name: 'Barba Completa',
      category: 'barba',
      price: 35.0,
      durationMinutes: 30,
      iconName: 'beard',
      description: 'Modelagem de barba com toalha quente, óleo hidratante e massagem facial.',
    ),
    const ServiceModel(
      id: 'srv_3',
      name: 'Design de Sobrancelha',
      category: 'sobrancelha',
      price: 20.0,
      durationMinutes: 15,
      iconName: 'eye',
      description: 'Limpeza e alinhamento de sobrancelha com navalha ou pinça.',
    ),
    const ServiceModel(
      id: 'srv_4',
      name: 'Combo Cabelo + Barba',
      category: 'combo',
      price: 70.0,
      durationMinutes: 60,
      iconName: 'combo',
      description: 'O serviço completo dos reis: corte personalizado e barba completa com toalha quente.',
    ),
  ];

  static final List<Plan> plans = [
    const Plan(
      id: 'plan_1',
      name: 'Plano Cavalheiro',
      price: 89.90,
      period: 'mensal',
      benefits: [
        'Até 2 cortes de cabelo por mês',
        'Reserva online com prioridade',
        '1 dose de Chopp cortesia por visita',
      ],
      recommended: false,
    ),
    const Plan(
      id: 'plan_2',
      name: 'Plano Barão',
      price: 139.90,
      period: 'mensal',
      benefits: [
        'Cortes de cabelo ilimitados',
        'Até 2 barbas por mês',
        'Chopp e água liberados',
        '10% de desconto em produtos',
      ],
      recommended: true,
    ),
    const Plan(
      id: 'plan_3',
      name: 'Plano Imperial',
      price: 199.90,
      period: 'mensal',
      benefits: [
        'Cabelo e barba ilimitados',
        'Sobrancelha inclusa',
        'Atendimento exclusivo VIP sem fila',
        'Bebidas liberadas no bar',
        '15% de desconto em produtos',
      ],
      recommended: false,
    ),
  ];

  static final Membership activeMembership = Membership(
    id: 'mem_1',
    userId: 'usr_1',
    planId: 'plan_2',
    planName: 'Plano Barão',
    startDate: '2026-06-07',
    endDate: '2026-07-07',
    status: 'active',
    remainingBenefits: [
      'Cortes de Cabelo Ilimitados',
      '1 Barba Restante (de 2)',
      '10% Desconto em Produtos',
    ],
    discountsUsed: 3,
    nextRenewalDate: '2026-07-07',
  );

  static final List<Appointment> appointments = [
    Appointment(
      id: 'apt_1',
      userId: 'usr_1',
      barberId: 'barb_1',
      barberName: 'Marcos Silva',
      barberAvatar: 'https://images.unsplash.com/photo-1517841905240-472988babdf9?q=80&width=150',
      services: [services[0]],
      date: '2026-07-08',
      time: '14:00',
      totalValue: 45.0,
      status: 'confirmed',
      notes: 'Gostaria de degradê baixo nas laterais.',
    ),
    Appointment(
      id: 'apt_2',
      userId: 'usr_1',
      barberId: 'barb_2',
      barberName: 'Arthur Santos',
      barberAvatar: 'https://images.unsplash.com/photo-1539571696357-5a69c17a67c6?q=80&width=150',
      services: [services[3]],
      date: '2026-06-20',
      time: '10:00',
      totalValue: 70.0,
      status: 'completed',
    ),
  ];

  static final List<NotificationModel> notifications = [
    NotificationModel(
      id: 'not_1',
      userId: 'usr_1',
      title: 'Lembrete de Agendamento',
      message: 'Seu corte de cabelo com Marcos Silva está agendado para amanhã às 14:00.',
      type: 'info',
      read: false,
      createdAt: '2026-07-07T18:00:00Z',
    ),
    NotificationModel(
      id: 'not_2',
      userId: 'usr_1',
      title: 'Assinatura Ativada',
      message: 'Parabéns, você agora é um membro oficial do Clube Cavalheiro (Plano Barão)!',
      type: 'success',
      read: true,
      createdAt: '2026-06-07T10:00:00Z',
    ),
  ];

  static final List<Review> reviews = [
    Review(
      id: 'rev_1',
      scheduleId: 'apt_2',
      userId: 'usr_1',
      userName: 'Fabio Zvir',
      barberId: 'barb_2',
      rating: 5,
      comment: 'Corte excelente e atendimento sensacional do Arthur. Barba impecável!',
      createdAt: '2026-06-20T11:15:00Z',
    ),
  ];
}
